app/lib/steps/logic.rb

```diff
@@ -43,7 +43,9 @@ def owns_vehicle?(session_data)
       def employed?(session_data)
         return false if asylum_supported?(session_data) || passported?(session_data)
 
-        ApplicantForm::EMPLOYED_STATUSES.map(&:to_s).include? session_data["employment_status"]
+        employment_status_form = FeatureFlags.enabled?(:self_employed) ? EmploymentStatusForm::EMPLOYED_STATUSES : ApplicantForm::EMPLOYED_STATUSES
```

Comment:
This variable should perhaps be called `employment_statuses` as it's an array of strings rather than a form.